homeassistant/components/upnp/__init__.py

```diff
@@ -150,8 +157,22 @@ def device_discovered(info: Mapping[str, Any]) -> None:
         model=device.model_name,
     )
 
+    update_interval_sec = entry.options.get(
+        CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+    )
+    update_interval = timedelta(seconds=update_interval_sec)
+    LOGGER.debug("update_interval: %s", update_interval)
+    coordinator = UpnpDataUpdateCoordinator(
+        hass,
+        device=device,
+        update_interval=update_interval,
+    )
+    device.coordinator = coordinator
```

Comment:
Don't add the coordinator to the device, the coordinator already holds the device